### PR TITLE
Problem: MLS member leaving/joining interaction with council nodes not documented (WIP #141)

### PR DESCRIPTION
**Pull request by tomtau**
Solution:
- revised the proposed design, so that joining is immediate (node fetches MLS handshakes directly from remote TDBE instead of waiting for commit message)
- sketched out basic types for NACK
- sketched out a state diagram and explanation of transitions with regards to different events and MLS handshake messages

### CODE REVIEW

File: docs/modules/tdbe.md [hunk 1]
```diff
@@ -19,10 +19,19 @@ Transaction Data Bootstrapping Enclave (TDBE):
 
 ## new/rejoining node
 (for non-genesis specified ones)
+- obtain old data (see below) from external node TDBE
+- request "invitation" from the external node TDBE:
```
Comment (yihuang) [inline docs/modules/tdbe.md:23]: it should request "invitation" from the left-most leaf node, right?

Comment (tomtau) [reply docs/modules/tdbe.md:23]: No, any node should do here, as add is triggered immediately by the newly joining node / operator (unlike remove which is triggered by commit of a block with some event / transaction)

Comment (tomtau) [reply docs/modules/tdbe.md:23]: it makes sense to request it from the external one it was fetching data from

Comment (tomtau) [reply docs/modules/tdbe.md:23]: any concerns if it is any node?

**Review by yihuang — APPROVED** [review]

File: docs/modules/tdbe.md [hunk 2]
```diff
@@ -19,10 +19,19 @@ Transaction Data Bootstrapping Enclave (TDBE):
...
+  - request should contain the merkle proof of the new node's associated staking stake + keypackage
```
Comment (devashishdxt) [inline docs/modules/tdbe.md]: What is `staking stake`?

**Review by devashishdxt — APPROVED** [review]

File: docs/modules/tdbe.md (hunk 2 above)
Comment (tomtau) [reply docs/modules/tdbe.md]: a typo, it should be "staking state"

**Pull request merged**